datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -1792,27 +1768,6 @@ mod tests {
         assert_eq!(simplify(expr), expected);
     }
 
-    #[test]
-    fn test_simplify_divide_zero_by_zero() {
```

Comment:
Can we please ensure this new ignore error behavior is covered in these unit tests?

Perhaps you could update `test_simplify_divide_by_zero` and instead expect that no rewrite occurs rather than deleting the entire test. 

Also, I think it is important to have a test that shows partial simplification occurs even when part of the expression errors

For example, perhaps we can add a test like:
```
(1 + 2) + (4/0) --> 3 + (4/0)
```

Reply:
Done. add such tests.